### DISCUSSION TITLE
`InAppPurchase` control

### DISCUSSION THREAD

**Discussion by Shawn91** [flet-dev/flet#5564] [Extensions]
It seems we currently can't turn a flet app into an ios/android app.

When it is possible to do so, will flet support in app purchase directly without requiring developers to know flutter? 

If the answer is yes,  is there a timeline for it?

Many thanks to the developing team's great work.

**Comment by ericgazoni** [comment]
I'm also interested in this feature, and I'd like to contribute it because that's something I'll really need in a near future. I looked at the contributor guidelines, but it does not really covers how to add something on the Dart/Flutter side and how to mirror it in Flet. If there is some code somewhere (like another integration) that I could use to get inspiration, I would probably be able to copy/paste my way through it.

**Comment by ndonkoHenri** [comment]
Solution to this seems to be only for mobile and MacOS, as per the first link below. I think the implementation of this functionality will come needy when [Flet goes mobile](https://flet.dev/blog/flet-mobile-update). Implementing it now will be a waste of time, I think.
Useful links: 
- https://pub.dev/packages/in_app_purchase
- https://codelabs.developers.google.com/codelabs/flutter-in-app-purchases#0

**Comment by ericgazoni** [comment]
> Solution to this seems to be only for mobile and MacOS, as per the first link below

It sure reads 
> This plugin supports in-app purchases (IAP) through an underlying store, which can be the App Store (on iOS and macOS) or Google Play (on Android)." 

Or am I missing something? 😅 

To clarify: I'm building mobile apps for iOS and Android. I don't think it would be a waste of time to include IAP in Flet at this stage, because I don't think it will automagically happen when Flet goes mobile, since it's not a hard requirement to have it implemented before we can actually build mobile packages. But for sure it will become a very demanded feature right afterwards.

If @FeodorFitsner has it already on the roadmap for the, then I agree they will probably implement it more efficiently than me, and I'll wait. 
Otherwise, I'm volunteering to implement the feature already now, because I don't have much alternatives anyway, I can't distribute my apps since they rely on subscriptions, threfore will be blocked by both app stores until they have IAP set up. Also, I feel if I can take a bit of work off @FeodorFitsner shoulders then they can focus on other stuff that matters 👍

**Comment by FeodorFitsner** [comment]
https://pub.dev/packages/in_app_purchase

**Comment by yucel685** [comment]
are there any studies on this topic?

**Comment by Master-Y0da** [comment]
Im currently interested on this topic...since I'm developing a flet app and I plan to use app purchase and subscriptions for both android and IOS

**Comment by jules-baratoux** [comment]
Has anyone successfully integrated the [`in_app_purchase`](https://pub.dev/packages/in_app_purchase) plugin into a Flet project yet? I’m specifically looking for a working implementation or a reference repo for iOS/Android subscriptions. Any guidance or examples from those who have explored this would be a huge help to the community. Thanks!
